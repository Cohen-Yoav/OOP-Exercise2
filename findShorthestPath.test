# findShortestPath method Testing file

CreateGraph graph1

CreateNode A 2
CreateNode B 1
CreateNode C 3
CreateNode D 1
CreateNode E 4

# check empty graph
FindPath graph1 A -> E

AddNode graph1 A
# check for sp from node to itself
FindPath graph1 A -> A

# check for sp where goal is not part of the graph
FindPath graph1 A -> E
AddNode graph1 B
AddEdge graph1 A B

# check standart sp from A to child B
FindPath graph1 A -> B

# check when not all goals are in graph
FindPath graph1 A -> E B

# check when not all starts are in graph
FindPath graph1 A E -> B
AddNode graph1 C
AddNode graph1 D
AddNode graph1 E

AddEdge graph1 A C
AddEdge graph1 B D
AddEdge graph1 B C
AddEdge graph1 C D
AddEdge graph1 C E
AddEdge graph1 D E

FindPath graph1 A -> E
FindPath graph1 A B C -> E D A
FindPath graph1 B C -> E D
