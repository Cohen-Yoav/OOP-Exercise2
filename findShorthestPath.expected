# findShortestPath method Testing file

created graph graph1

created node A with cost 2
created node B with cost 1
created node C with cost 3
created node D with cost 1
created node E with cost 4

# check empty graph
shortest path in graph1:

added node A to graph1
# check for sp from node to itself
shortest path in graph1: A

# check for sp where goal is not part of the graph
shortest path in graph1:
added node B to graph1
added edge from A to B in graph1

# check standart sp from A to child B
shortest path in graph1: A B

# check when not all goals are in graph
shortest path in graph1: A B

# check when not all starts are in graph
shortest path in graph1: A B
added node C to graph1
added node D to graph1
added node E to graph1

added edge from A to C in graph1
added edge from B to D in graph1
added edge from B to C in graph1
added edge from C to D in graph1
added edge from C to E in graph1
added edge from D to E in graph1

shortest path in graph1: A B D E
shortest path in graph1: A
shortest path in graph1: B D
